crates/spidermonkey-embedding-splicer/src/bindgen.rs

```diff
@@ -363,9 +369,38 @@ impl JsBindgen<'_> {
         return intrinsic.name().to_string();
     }
 
-    fn exports_bindgen(&mut self) {
+    fn exports_bindgen(&mut self, guest_exports: &Vec<String>) {
         for (key, export) in &self.resolve.worlds[self.world].exports {
             let name = self.resolve.name_world_key(key);
+
+            // Do not generate exports when the guest export is not implemented.
+            // We check both the full interface name - "ns:pkg@v/my-interface" and the
+            // aliased interface name "myInterface". All other names are always
+            // camel-case in the check.
+            match key {
+                WorldKey::Interface(iface) => {
+                    if !guest_exports.contains(&name) {
+                        let iface = &self.resolve.interfaces[*iface];
+                        if let Some(name) = iface.name.as_ref() {
+                            let camel_case_name = name.to_lower_camel_case();
+                            if !guest_exports.contains(&camel_case_name) {
+                                continue;
+                            }
+                            // TODO: move populate_export_aliases to a preprocessing
+                            // step that doesn't require esm_bindgen, so that we can
+                            // do alias deduping here as well.
```

Comment:
This may bring in possibly unused interfaces in some alias conflict cases, which would require the refactoring as discussed around alias preprcessing. I'm okay to land this without that to start.